File: 1-100/twoSum/twoSum.py
def twoSum(nums: List[int], target: int) -> List[int]:
    dict = {}
    for i in range(len(nums)):
        if nums[i] in dict:
            if i * 2 == target:
                return [dict[nums[i]], i]
        elif target - nums[i] in dict:
            return [dict[target - nums[i]], i]
        else:
            dict[nums[i]] = i


# 按间距中的绿色按钮以运行脚本。
if __name__ == '__main__':
    print(twoSum([2, 7, 11, 15], 9))

File: 1-100/twoSum/TwoSum.scala
object TwoSum {
  def main(args: Array[String]): Unit = {
    val nums = Array[Int](3,3)
    val target = 6
    twoSum(nums, target).foreach(println)
  }

  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    var mp = scala.collection.mutable.Map[Int, Int]()
    for ((value, key) <- nums.zipWithIndex) {
      mp.get(target - value) match {
        case Some(x) => return (Array(x, key))
        case None => mp += (value -> key)
      }
    }
    Array(0, 0)
  }
}


